assets/js/components/Dashboard/NewPaMessage/NewPaMessage.tsx

```diff
@@ -45,7 +46,36 @@ const NewPaMessage = () => {
     setVisualText(alertMessage);
   };
 
-  const onImportLocations = () => {};
+  const onImportLocations = (informedEntities: InformedEntity[]) => {
+    const importedSigns = informedEntities.reduce((acc, entity) => {
+      const informedPlaces = entity.stop
+        ? places.filter((place) => place.id === entity.stop)
+        : places;
+
+      if (entity.route) {
+        let signsToAdd = informedPlaces
+          .flatMap((place) => place.screens)
+          .filter((screen) =>
+            getRouteIdsForSign(screen).includes(entity.route ?? ""),
```

Comment:
**suggestion:** If you create a variable of the narrowed `entity.route` value you won't need that nullish coalescing operator.

```suggestion
      if (entity.route) {
        const entityRoute = entity.route
        let signsToAdd = informedPlaces
          .flatMap((place) => place.screens)
          .filter((screen) =>
            getRouteIdsForSign(screen).includes(entityRoute),
```